### PR TITLE
Updates to Actions Pipeline

### PR DESCRIPTION
**Pull request by Voark**
## Summary
- Added deploy to Github pages action (code taken from https://github.com/icssc/fellowship/blob/main/.github/workflows/deploy.yml)
- Removed staging on label, making it mandatory for every new PR. This is to account for the fact that changes will be deployed to prod automatically by the "Deploy to Github Pages" again, so we should make sure that everything is OK. Also makes staging cleaner as no labels and doesn't require educating outside contributors on how to stage PR.
- Remove environment delete action as default Github Token does not have perms to delete an environment
- Moved CDK code into this repo instead of having it spread across multiple repos

## Test Plan
- https://github.com/icssc/AntAlmanac/pull/349 (TLDR the deploy and close actions worked there with no issue)
- This PR
- Cannot test deploy to pages action without merging to main

## Issues
- Still no auto environment delete (need to use PAT with permissions instead of default token) so environments will need to be cleaned manually

### CODE REVIEW

**Comment by Voark** [comment]
As can be seen, the deploy skips when label is applied
image

**Comment by ChaseC99** [comment]
Feel free to add me as a reviewer when you're ready!

**Comment by ChaseC99** [comment]
Is this ready for review?

**Comment by Voark** [comment]
Yeah sorry, didn't see the notification for the first comment.

File: .github/workflows/deploy_pages.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
```
Comment (ChaseC99) [inline .github/workflows/deploy_pages.yaml:6]: So excited for this too!

File: .github/workflows/stage_pr.yaml [hunk 2]
```diff
@@ -2,8 +2,7 @@ name: "Deploy Pull Request"
 
 on: 
   pull_request:
-    branches: main
-    types: [labeled, opened, synchronize]
+      types: [opened, synchronize]
```
Comment (ChaseC99) [inline .github/workflows/stage_pr.yaml:5]: What is synchronize?

**Review by ChaseC99 — APPROVED** [review]
LGTM! Great work on all this 💯

**Comment by ChaseC99** [comment]
> Yeah sorry, didn't see the notification for the first comment.

I didn't realize I sent 2 comments back-to-back. I sent the first one last night and then forgot about it in the morning. Didn't mean to spam you lol.

File: .github/workflows/stage_pr.yaml (hunk 2 above)
Comment (Voark) [reply .github/workflows/stage_pr.yaml:5]: Synchronize is on branch update, so when new code is pushed to the branch.

**Comment by Voark** [comment]
> > Yeah sorry, didn't see the notification for the first comment.
> 
> I didn't realize I sent 2 comments back-to-back. I sent the first one last night and then forgot about it in the morning. Didn't mean to spam you lol.

No worries, I wouldn't even have seen it if not for the second one.

**Pull request merged**